lib/block_view_types.go

```diff
@@ -1991,8 +2127,32 @@ type MessagingGroupEntry struct {
 	isDeleted bool
 }
 
+// NewMessagingGroupEntry creates a new MessagingGroupEntry object.
+func NewMessagingGroupEntry(groupOwnerPublicKey *PublicKey, messagingPublicKey *PublicKey, messagingGroupKeyName *GroupKeyName,
+	messagingGroupMembers []*MessagingGroupMember, extraData map[string][]byte, blockHeight uint64) *MessagingGroupEntry {
+
+	messagingGroupEntry := MessagingGroupEntry{
+		GroupOwnerPublicKey:   groupOwnerPublicKey,
+		MessagingPublicKey:    messagingPublicKey,
+		MessagingGroupKeyName: messagingGroupKeyName,
+		MessagingGroupMembers: messagingGroupMembers,
+		ExtraData:             extraData,
+	}
+
+	if MigrationTriggered(blockHeight, DeSoUnlimitedDerivedKeysAndMessageMutingAndMembershipIndex) {
```

Comment:
hmm, I don't love using `MigrationTriggered` to handle regular block height logic. Till now, we used `MigrationTriggered` only in DeSoEncoder logic